test/support/protocol_utils.rb

```diff
@@ -230,6 +230,29 @@ def assert_locals_result expected, frame_idx: 0
       end
     end
 
+    def assert_threads_result(expected_names)
+      case ENV['RUBY_DEBUG_TEST_UI']
+      when 'vscode'
+        res = send_dap_request 'threads'
+        failure_msg = FailureMessage.new{create_protocol_message "result:\n#{JSON.pretty_generate res}."}
+
+        threads = res.dig(:body, :threads)
+
+        assert_equal expected_names.count, threads.count, failure_msg
+
+        thread_names = threads.map { |t| t[:name] }
+
+        expected_names.each do |expected|
+          thread_names.reject! do |name|
+            name.match?(expected)
+          end
+        end
+
+        failure_msg = FailureMessage.new{create_protocol_message "result:\n#{JSON.pretty_generate res}.\nExpect all thread names to be matched. Unmatched threads:"}
+        assert_equal [], thread_names, failure_msg
```

Comment:
Can we use `assert_empty`?

Reply:
I decided to use `assert_equal` because it provides better highlighting in the failure messages:

<img width="80%" alt="截圖 2022-04-17 10 25 01" src="https://user-images.githubusercontent.com/5079556/163708602-49ba06bf-63ea-47ba-b25a-b3b213c6185c.png">


**`assert_empty`**

<img width="90%" alt="截圖 2022-04-17 10 25 15" src="https://user-images.githubusercontent.com/5079556/163708606-aeb69d65-a12f-42a9-903e-39f40df2f29d.png">

I guess it's because `assert_empty` doesn't care about individual items in the array, but `assert_equal` does. So it provides a more detailed information.